### PR TITLE
Add metadata and return_job to job dict

### PR DESCRIPTION
**Pull request by pass-by-value**

### CODE REVIEW

**Comment by whiteinge** [comment]
@pass-by-value I'm seeing a traceback on `schedule.save`. I'm also not seeing the scheduler actually running the job or sending results up to the master.

```python
Traceback (most recent call last):
  File "/Users/shouse/src/salt/salt/salt/minion.py", line 1068, in _thread_return
    return_data = func(*args, **kwargs)
  File "/Users/shouse/src/salt/salt/salt/modules/schedule.py", line 563, in save
    schedule = list_(return_yaml=False)
  File "/Users/shouse/src/salt/salt/salt/modules/schedule.py", line 80, in list_
    if schedule[job][item] == 'true':
KeyError: 'return_job'
```

**Comment by pass-by-value** [comment]
@whiteinge What schedules are added to your setup currently? Could you do a ``salt \* schedule.list show_all=True``?

File: salt/modules/schedule.py [hunk 1]
```diff
@@ -222,6 +222,12 @@ def build_schedule_item(name, **kwargs):
         if item in kwargs:
             schedule[name][item] = kwargs[item]
 
+    if 'return_job' in kwargs:
+        schedule[name]['return_job'] = True
```
Comment (thatch45) [inline salt/modules/schedule.py]: should this be:

```
schedule[name]['return_job'] = kwargs['return_job']
```

?

Comment (pass-by-value) [inline salt/modules/schedule.py]: Updated. Thanks!

**Pull request merged**